src/fs.rs

```diff
@@ -42,11 +42,11 @@ pub fn cp(src: impl AsRef<Path>, dst: impl AsRef<Path>) -> Result<()> {
 }
 fn _cp(src: &Path, dst: &Path) -> Result<()> {
     let _guard = gsl::read();
-    let mut dst = dst;
     let mut _tmp;
+    let mut dst = dst;
```

Comment:
Making `mut dst: &Path` didn't work, because of the lifetime issue where _tmp didn't live long enough.